src/platform/datapath_raw_xdp_win.c

```diff
@@ -624,39 +667,80 @@ CxPlatDpRawInterfaceInitialize(
             goto Error;
         }
 
-        Status =
-            XskSetSockopt(
-                Queue->TxXsk, XSK_SOCKOPT_TX_RING_SIZE, &Xdp->TxRingSize, sizeof(Xdp->TxRingSize));
+        Flags = XSK_BIND_FLAG_TX;
+        Status = XskBind(Queue->TxXsk, Interface->ActualIfIndex, i, Flags);
         if (QUIC_FAILED(Status)) {
             QuicTraceEvent(
                 LibraryErrorStatus,
                 "[ lib] ERROR, %u, %s.",
                 Status,
-                "XskSetSockopt(XSK_SOCKOPT_TX_RING_SIZE)");
+                "XskBind");
             goto Error;
         }
 
+        //
+        // Before enabling the TX offload descriptors, check whether the offload
+        // is currently enabled on the interface. If it isn't, assume the
+        // interface (or XDP driver) does not and will not support the offload.
+        // If the offload is disabled (and perhaps re-enabled) later, it will
+        // get picked up by the XskRingOffloadChanged check in the TX data path.
+        //
+        uint32_t TxChecksumOffload = FALSE;
+        Status = GetTxOffloadConfig(Queue, &TxChecksumOffload);
+        if (QUIC_FAILED(Status)) {
+            CXPLAT_DBG_ASSERT(!TxChecksumOffload);
+            QuicTraceEvent(
+                LibraryErrorStatus,
+                "[ lib] ERROR, %u, %s.",
+                Status,
+                "GetTxOffloadConfig(Queue, &TxChecksumOffload)");
+        }
+
+        if (TxChecksumOffload) {
+            Status =
+                XskSetSockopt(
+                    Queue->TxXsk, XSK_SOCKOPT_TX_OFFLOAD_CHECKSUM, &TxChecksumOffload,
+                    sizeof(TxChecksumOffload));
+            if (QUIC_SUCCEEDED(Status)) {
+                Queue->OffloadStatus.Transmit.ChecksumOffload = TRUE;
+                Queue->OffloadStatus.Transmit.ChecksumOffloadExtensions = TRUE;
+            } else {
+                QuicTraceEvent(
+                    LibraryErrorStatus,
+                    "[ lib] ERROR, %u, %s.",
+                    Status,
+                    "XskSetSockopt(XSK_SOCKOPT_TX_OFFLOAD_CHECKSUM)");
+            }
+        } else {
+            QuicTraceEvent(
```

Comment:
This is a duplicate log in the QUIC_FAILED case above.

Reply:
I'm not sure - `GetTxOffloadConfig` can return QUIC_SUCCEEDED and `TxChecksumOffload` can be false.